internal/server/iigointernal/judiciary.go

```diff
@@ -410,3 +528,23 @@ func searchEvalReturnForRuleName(name string, evaluationReturn roles.EvaluationR
 	}
 	return false
 }
+
+// mergeEvalResults takes two evaluation results and merges them to provide a unified set
+func mergeEvalResults(set1 map[shared.ClientID]roles.EvaluationReturn, set2 map[shared.ClientID]roles.EvaluationReturn) map[shared.ClientID]roles.EvaluationReturn {
+	for key, val := range set1 {
+		if set2Val, ok := set2[key]; ok {
+			defRules := set2Val.Rules
+			defEvals := set2Val.Evaluations
+			finalRules := append(defRules, val.Rules...)
+			finalEvals := append(defEvals, val.Evaluations...)
```

Comment:
this is definitely going to bite you--if the `defRules` container has enough space, `defRules` itself is mutated.

https://tour.golang.org/moretypes/15

this would immediately surface upon a small unit test.

Reply:
Thanks for the catch, would never have known. Patched in latest commit